include/itkCudaImageFromImageFilter.h

```diff
@@ -0,0 +1,105 @@
+ *
+ *  Copyright NumFOCUS
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *         https://www.apache.org/licenses/LICENSE-2.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *=========================================================================*/
+#ifndef itkCudaImageFromImageFilter_h
+#define itkCudaImageFromImageFilter_h
+
+#include "itkImageToImageFilter.h"
+#include "itkCudaImage.h"
+
+namespace itk
+{
+
+/** \class CudaImageFromImageFilter
+ * \brief Adaptor filter producing a CudaImage that shares the input host pixel buffer.
+ *
+ * This filter takes an arbitrary itk::Image as input and produces an itk::CudaImage
+ * of the same pixel type and dimension without copying pixel data. The output's
+ * PixelContainer pointer is set to the input's container enabling zero‑copy
+ * transition into the CUDA pipeline. Meta-information (spacing, origin, direction,
+ * and regions) is propagated. No device (GPU) upload is forced; that occurs lazily
+ * when downstream CUDA operations require it.
+ *
+ * Lifetime: The caller must ensure the input image (and thus its PixelContainer)
+ * outlives the filter output. Modifying the input buffer modifies the output.
+ * If isolation is required, insert a copier filter before this adaptor.
+ *
+ * \ingroup ITKCudaCommon
+ */
+template <typename TInputImage,
+          typename TOutputImage =
+            itk::CudaImage<typename TInputImage::PixelType, TInputImage::ImageDimension>>
+class ITK_TEMPLATE_EXPORT CudaImageFromImageFilter
+  : public ImageToImageFilter<TInputImage, TOutputImage>
+{
+public:
+  ITK_DISALLOW_COPY_AND_MOVE(CudaImageFromImageFilter);
+
+  using Self = CudaImageFromImageFilter;
+  using Superclass = ImageToImageFilter<TInputImage, TOutputImage>;
+  using Pointer = SmartPointer<Self>;
+  using ConstPointer = SmartPointer<const Self>;
+
+  itkNewMacro(Self);
+  itkTypeMacro(CudaImageFromImageFilter, ImageToImageFilter);
+
+  using InputImageType = TInputImage;
+  using OutputImageType = TOutputImage;
+
+protected:
+  CudaImageFromImageFilter() = default;
+  ~CudaImageFromImageFilter() override = default;
+
+  void GenerateOutputInformation() override
+  {
+    Superclass::GenerateOutputInformation();
+    if (const auto * in = this->GetInput())
+    {
+      auto * out = this->GetOutput();
+      out->CopyInformation(in);
+    }
+  }
+
+  void AllocateOutputs() override
+  {
+    // Intentionally skip default allocation: we will share the host pixel container.
+    // The regions are finalized in GenerateData.
+  }
```

Comment:
I wonder why this is required as `AllocateOutputs` is not called if you overload `GenerateData`.